src/components/data_comps_mct/datm/cime_config/buildnml

```diff
@@ -44,18 +44,30 @@ def _create_namelists(case, confdir, inst_string, infile, nmlgen, data_list_path
     datm_co2_tseries = case.get_value("DATM_CO2_TSERIES")
     atm_grid = case.get_value("ATM_GRID")
     grid = case.get_value("GRID")
-    clm_usrdat_name = case.get_value("CLM_USRDAT_NAME")
+    if get_model() != "e3sm":
+        clm_usrdat_name = case.get_value("CLM_USRDAT_NAME")
+    else:
+        elm_usrdat_name = case.get_value("ELM_USRDAT_NAME")
 
     #----------------------------------------------------
     # Check for incompatible options.
     #----------------------------------------------------
-    if "CLM" in datm_mode and case.get_value("COMP_LND") == "clm":
-        expect(datm_presaero != "none",
-               "A DATM_MODE for CLM is incompatible with DATM_PRESAERO=none.")
-        expect(datm_topo != "none",
-               "A DATM_MODE for CLM is incompatible with DATM_TOPO=none.")
-    expect(grid != "CLM_USRDAT" or clm_usrdat_name in ("", "UNSET"),
-           "GRID=CLM_USRDAT and CLM_USRDAT_NAME is NOT set.")
+    if get_model() != "e3sm":
```

Comment:
The code looks identical in both branches with the only difference being ELM vs CLM. You should capture this difference in a single string so the code can be reused.